File: app/src/main/java/ua/com/foxminded/locationtrackera/ui/auth/reset/ResetPasswordContract.java
package ua.com.foxminded.locationtrackera.ui.auth.reset;

import ua.com.foxminded.locationtrackera.mvi.FragmentContract;
import ua.com.foxminded.locationtrackera.ui.auth.reset.state.ResetPasswordScreenEffect;
import ua.com.foxminded.locationtrackera.ui.auth.reset.state.ResetPasswordScreenState;

public class ResetPasswordContract {

    public interface ViewModel extends FragmentContract.ViewModel<ResetPasswordScreenState, ResetPasswordScreenEffect> {
        void resetPassword(String email);
    }

    public interface View extends FragmentContract.View {
        void setProgressVisibility(boolean isProgressVisible);

        void showToastMessage(int idStringResource);

        void showEmailError(int emailError);
    }

    public interface Host extends FragmentContract.Host {
    }
}

File: app/src/main/java/ua/com/foxminded/locationtrackera/ui/auth/reset/ResetPasswordContract.kt
package ua.com.foxminded.locationtrackera.ui.auth.reset

import ua.com.foxminded.locationtrackera.mvi.fragments.FragmentContract
import ua.com.foxminded.locationtrackera.ui.auth.reset.state.ResetPasswordScreenEffect
import ua.com.foxminded.locationtrackera.ui.auth.reset.state.ResetPasswordScreenState

class ResetPasswordContract {
    interface ViewModel : FragmentContract.ViewModel<
            ResetPasswordScreenState,
            ResetPasswordScreenEffect> {
        fun resetPassword(email: String?)
    }

    interface View : FragmentContract.View {
        fun setProgressVisibility(isProgressVisible: Boolean)
        fun showToastMessage(idStringResource: Int)
        fun showEmailError(emailError: Int)
    }

    interface Host : FragmentContract.Host
}
